File: backend_spring_boot/src/main/java/com/stage/paiement/dto/EcheancierRequest.java
package com.stage.paiement.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class EcheancierRequest {

    @NotNull
    private String nomClient;

    @Min(1)
    private double montantTotal;

    @Min(1)
    private int nombreMensualites;
}

File: paiement_facilite_app/lib/models/echeancier_request.dart
class EcheancierRequest {
  final int clientId;
  final double montantTotal;
  final int nombreMensualites;

  EcheancierRequest({
    required this.clientId,
    required this.montantTotal,
    required this.nombreMensualites,
  });

  Map<String, dynamic> toJson() {
    return {
      "clientId": clientId,
      "montantTotal": montantTotal,
      "nombreMensualites": nombreMensualites,
    };
  }
}
